### ISSUE TITLE
Stream still playing when stopping it (KODI 19)

### ISSUE THREAD

**Issue by ghost** [kodi-pvr/pvr.iptvsimple#501]
When I stop a stream it still plays back according to the interface of my device using KODI 19.
I tried my SAT>IP server (Fritz!WLAN Repeater DVB-C).
With VLC and DVBViewer Pro I don't have this issue, but I want to use KODI instead.
Problem is, when I try 3 different streams, the device is locked because both tuners are already in use.
Can this be fixed please? Looks like a bug.

This is an example of my stream URL:
rtsp://192.168.0.4:554?freq=298&msys=dvbc&sr=6900&mtype=256qam&pids=0,16,17,20,3200,3201,3211

**Comment by ghost** [comment]
I just tried KODI 15.2 and this one hasn't this issue. Will try other versions as well and report it back here.

KODI 16.1 also works fine.
KODI 17.5 also works fine.

Problem occurs for the first time with KODI 18.0.

**Comment by phunkyfish** [comment]
We need the latest version of ffmpeg for this to work but it’s not released yet.

**Comment by phunkyfish** [comment]
Is this a satip stream?

**Comment by phunkyfish** [comment]
Can you try adding this to one of the channels and seeing it works:

```
#KODIPROP:inputstream=inputstream.ffmpegdirect
#KODIPROP:inputstream.ffmpegdirect.open_mode=ffmpeg
#KODIPROP:inputstream.ffmpegdirect.is_realtime_stream=true
```

Then also change the url to start with `satip://` instead of `rtsp://`.

**Comment by ghost** [comment]
> We need the latest version of ffmpeg for this to work but it’s not released yet.

Okay, I see. I hope they will release it soon.

**Comment by ghost** [comment]
> Is this a satip stream?

Yes it is.

**Comment by ghost** [comment]
> Can you try adding this to one of the channels and seeing it works:
> 
> ```
> #KODIPROP:inputstream=inputstream.ffmpegdirect
> #KODIPROP:inputstream.ffmpegdirect.open_mode=ffmpeg
> #KODIPROP:inputstream.ffmpegdirect.is_realtime_stream=true
> ```
> 
> Then also change the url to start with `satip://` instead of `rtsp://`.

Sweet, yes, this works for now... thanks!

**Comment by phunkyfish** [comment]
Did you change the URL to satip:// or just used the same URL?

**Comment by ghost** [comment]
It tried both, and both do work.

Edit: I tried again, but now only RTSP does work, anyway no problem.

**Comment by ghost** [comment]
But these lines have to be added to each channel, not only on top of the M3U file, right?

#KODIPROP:inputstream=inputstream.ffmpegdirect
#KODIPROP:inputstream.ffmpegdirect.open_mode=ffmpeg
#KODIPROP:inputstream.ffmpegdirect.is_realtime_stream=true

**Comment by phunkyfish** [comment]
Yes, it’s a workaround until Kodi 19.1 is out.

**Comment by phunkyfish** [comment]
I'll close this issue for now the fix is now also scheduled for Kodi 19.1
